pkg/algorithm/topology_aware_scheduler.go

```diff
@@ -44,9 +45,8 @@ type topologyAwareScheduler struct {
 	// because guaranteed pods can avoid preempting opportunistic pods only among buddy cells (this is decided
 	// by the buddy cell allocation algorithm).
 	crossPriorityPack bool
-	// whether or not the scheduler should avoid using nodes that are not suggested by K8s.
-	// should be true when the scheduler is used for scheduling physical GPUs (i.e., for opportunistic pods)
-	considerSuggestedNodes bool
+	// whether or not the scheduler should prefer using nodes that are healthy and suggested by K8s
+	preferHealthyAndSuggestedNodes bool
 }
```

Comment:
Maybe we can decouple unsuggested and bad nodes, in all places, because, anyway unsuggested need to be a flag that supports to be disabled .